feat : 예약 생성 api 구현

heejintv opened the pull request:
## #️⃣ Issue Number

## 📝 요약(Summary)

[예약 생성 API 구현]
- 예약 엔티티 및 레포지토리 구현
- 예약 상태 (Status) 변경 메소드 구현
- 예약 컨트롤러, 서비스 뼈대 생성
- 예약 생성(사용자가 예약을 생성함) API 구현
- ReservationService 수정 : 기존의 ReservationRepository 의존성을 제거하고, ReservationReader와 ReservationWriter를 주입받아 사용하도록 변경.

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)

## 💬 공유사항 to 리뷰어

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

heejintv committed:
[FEATURE] Reservation 엔티티 및 repository 생성

- User 엔티티의 pk 'id'를 fk로 사용
- 'shopId'와 'ownerUserId' 모두 현재는 원장님의 User.id 저장
- 향후 Shop 엔티티 만들어진다면 'shopId'는 Shop.id를, 'ownerUserId'는 원장님의 User.id 저장할 것.
- Status(예약 상태)는 총 대기, 확정, 취소, 거절로 정의함.
- Status의 초기 상태 : PENDING(대기)

heejintv committed:
[REFACTOR] 예약 상태 변경 메소드 구현

- 예약 상태 변경 메소드 추가
- confirm(), reject(), cancel()
- 확정 / 거절 : 대기(PENDING) 상태일 때만 가능
- 취소 : 대기(PENDING) or 확정(CONFIRMED) 상태일 때 가능

heejintv committed:
[FEATURE] 고객 예약 생성 API 구현

- 고객 예약 생성 API의 컨트롤러, 서비스 로직 구현
- POST /api/reservations 구현
- Principal.getName()을 UserService로 전달하여 DB PK(User.id)로 변환하도록 함.

heejintv committed:
Merge branch 'develop' into feature/heejin/#23

seobeeeee1001 commented on src/main/java/com/example/easybooking/reservation/service/ReservationService.java:
```diff
@@ -0,0 +1,54 @@
+        );
+
+        Reservation savedReservation = reservationRepository.save(newReservation);
+        return new ReservationResponse(savedReservation);
```
레이어 구성이 지금 컨트롤러 - 서비스 - 구현 - 레포지토리로 돼있어요. 
그래서 서비스에서는 구현 레이어의 클래스를 사용해서 데이터를 읽고 쓰도록 하려고 합니다. 
UserReader나 UserWriter 등의 구현 레이어 클래스가 있으니 참고해주시면 감사하겠습니다.
코드 짤 때 좀 복잡할 순 있지만 비즈니스 로직이 명확해지고, 확장성이 좋아진다는 점이 있어서 이번 프로젝트에 초기부터 도입하게 됐습니다. 수고스럽겠지만 수정 부탁드립니다..!!

seobeeeee1001 commented on src/main/java/com/example/easybooking/reservation/domain/Reservation.java:
```diff
@@ -0,0 +1,79 @@
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Reservation {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private Long shopId;       // 현재는 원장님 User.id (추후 Shop Id로 확장 가능성)
+
+    @Column(nullable = false)
+    private Long ownerUserId;  // 담당 원장님 User.id (현재 shopId와 동일)
+
+    @Column(nullable = false)
+    private Long customerId;   // 고객 User.id
+
+    // --- 예약 정보 ---
+    @Column(nullable = false)
+    private LocalDate date;
+
+    @Column(nullable = false)
+    private LocalTime time;
+
+    private String designImageURL;
+
+    @Column(nullable = false, updatable = false)
+    private LocalDateTime createdAt;
+
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private Status status;     // 예약 상태
```
그리고 예약 정보들은 더 고민해봐야할 것 같아요

매장마다 예약 폼 옵션들이 다를 수도 있어서 그걸 어떻게 받아낼지 생각해봐야할 것 같습니다

seobeeeee1001 reviewed (commented):
수정 사항이 좀 있을 것 같지만 예약 관련 처리 잘 해주셨네요. 고생 많으셨습니다! 
근데 pr이 중복된게 있는 것 같은데 맞으실까요?
처리해주시면 감사하겠습니다.

그리고 pr 컨벤션은 feat : 예약 생성 api 구현 
이런 식으로 수정해주시고, asignee, label 등 적절하게 선택해주세요!

heejintv committed:
[REFACTOR] 예약 Service 레이어 구현 계층(Reader/Writer) 추가

Reservation 도메인의 데이터 접근 방식 수정
- ReservationReader 추가
- ReservationWriter 추가
- ReservationService 수정 : 기존의  ReservationRepository 의존성을 제거하고, ReservationReader와 ReservationWriter를 주입받아 사용하도록 변경.

heejintv committed:
Merge pull request #1 from WAP-SNAPBOOK/develop

Develop

heejintv committed:
[FIX] 예약 신청 시 인증 사용자 조회 오류 개선

[예약 및 인증 로직 수정]
- ReservationController 및 ReservationService 로직을 ProviderId 조회에서 인증된 사용자의 내부 DB PK(userId)를 직접 사용하는 방식으로 변경
